### PR TITLE
[TRB-35513]: Update probe registration to create SLO action merge specs

### PR DESCRIPTION
**Pull request by libai98**
I have merged this PR https://github.com/turbonomic/turbo-go-sdk/pull/153 in turbo-go-sdk  to updated the go bindings for ActionMergePolicyDTO and provide some builder util method for horizontal scale action merge policy DTO.
This PR is to utilize those util methods to update kubeturbo probe registration to create SLO action merge specs.

 Developer Checks
 [x] Full build with unit tests and Checkstyle / SpotBugs tests
 [x] Unit tests added / updated
 [x]No unlicensed images, no third-party code (such as from StackOverflow)
      Integration (Robot Framework) tests added / updated
      Manual testing done (and described)
      Product sweep run and passed
      Developer wiki updated (and linked to this description)

### CODE REVIEW

File: pkg/registration/registration_client.go [hunk 1]
```diff
@@ -228,11 +228,47 @@ func (rClient *K8sRegistrationClient) GetActionMergePolicy() []*proto.ActionMerg
...
+			},
+		},
+	}
+
+	containerHorizontalScaleMerge := builder.NewHorizontalScaleMergeSpecBuilder().
+		ForEntityType(proto.EntityDTO_CONTAINER_POD).
+		ForCommodity(proto.CommodityDTO_VCPU).
+		ForCommodity(proto.CommodityDTO_VMEM).
+		ForCommodity(proto.CommodityDTO_VCPU_REQUEST).
+		ForCommodity(proto.CommodityDTO_VMEM_REQUEST).
```
Comment (ading1977) [inline pkg/registration/registration_client.go:264]: Horizontal scale action are not triggered by resource commodities today. They are triggered by SLO commodities like ResponseTime and Transactions.

File: pkg/registration/registration_client.go [hunk 2]
```diff
@@ -228,11 +228,47 @@ func (rClient *K8sRegistrationClient) GetActionMergePolicy() []*proto.ActionMerg
...
+
+	containerPodDataDaemonSet := proto.EntityDTO_ContainerPodData{
+		ControllerData: &proto.EntityDTO_WorkloadControllerData{
+			ControllerType: &proto.EntityDTO_WorkloadControllerData_DaemonSetData{
+				DaemonSetData: &proto.EntityDTO_DaemonSetData{
+					// Populate the fields specific to DaemonSetData
+				},
+			},
+		},
+	}
```
Comment (ading1977) [inline pkg/registration/registration_client.go:247]: Why do we specify daemonset here? There won't be SLO horizontal scale actions for daemonset.

File: pkg/registration/registration_client.go [hunk 3]
```diff
@@ -228,11 +228,47 @@ func (rClient *K8sRegistrationClient) GetActionMergePolicy() []*proto.ActionMerg
...
+
+	containerPodDataReplicaSet := proto.EntityDTO_ContainerPodData{
+		ControllerData: &proto.EntityDTO_WorkloadControllerData{
+			ControllerType: &proto.EntityDTO_WorkloadControllerData_ReplicaSetData{
+				ReplicaSetData: &proto.EntityDTO_ReplicaSetData{
+					// Populate the fields specific to ReplicaSetData
+				},
```
Comment (ading1977) [inline pkg/registration/registration_client.go:254]: Why replicaSet? Shouldn't this be deployment?

File: pkg/registration/registration_client.go (hunk 2 above)
Comment (libai98) [reply pkg/registration/registration_client.go:247]: This is specified for the filter to not have the merged action: https://vmturbo.atlassian.net/wiki/spaces/AE/pages/3668901989/SLO+Horizontal+Scale+Action+Merging#Replicas-Represented-as-Attribute-on-WorkloadController
DaemonSets and ReplicaSets
Actions related to DaemonSets and ReplicaSets will not be merged. We should ignore any Provision/Suspend actions related to them during the merge process.

File: pkg/registration/registration_client.go (hunk 3 above)
Comment (libai98) [reply pkg/registration/registration_client.go:254]: Please see the comments above.

File: pkg/registration/registration_client.go (hunk 1 above)
Comment (libai98) [reply pkg/registration/registration_client.go:264]: Yes, thanks! I will make the change.

**Pull request closed without merging**